### PR TITLE
{devel}[GCCcore/12.3.0] Qt5 v5.15.10

### PR DESCRIPTION
**Pull request by stefan-wolfsheimer**
(created using `eb --new-pr`)

Depends on [#18538](https://github.com/easybuilders/easybuild-easyconfigs/pull/18538)
And on [#18540](https://github.com/easybuilders/easybuild-easyconfigs/pull/18540)

### CODE REVIEW

File: easybuild/easyconfigs/q/Qt5/Qt5-5.15.10_fix-qtwebengine-breakpad-glibc-2.34.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc	2023-08-14 14:19:20.751150000 +0200
```
Comment (casparvl) [inline easybuild/easyconfigs/q/Qt5/Qt5-5.15.10_fix-qtwebengine-breakpad-glibc-2.34.patch:1]: Please add a patch description. Preferably include what error you get without the patch, how this fixes it, and referencing upstream issues if there are any. That should help us determine for future versions if the patch is still needed (e.g. by checking if the upstream issue is still open, or is merged with references to a fix).

**Review by casparvl — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/q/Qt5/Qt5-5.15.10_fix-qtwebengine-breakpad-glibc-2.34.patch (hunk 1 above)
Comment (stefan-wolfsheimer) [reply easybuild/easyconfigs/q/Qt5/Qt5-5.15.10_fix-qtwebengine-breakpad-glibc-2.34.patch]: I have submitted the description to my branch. Do I need to do something to retrigger the test pipeline with?

Comment (casparvl) [reply easybuild/easyconfigs/q/Qt5/Qt5-5.15.10_fix-qtwebengine-breakpad-glibc-2.34.patch]: Nope, CI retriggers automatically after a push to the feature branch in this PR. Thanks!

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
@stefan-wolfsheimer you forgot to update the checksum for the patch after adding the patch description (happens to me all the time as well, so easy to forget :D)

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.7, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/039d3d98f716c8fc4130944dd3034880 for a full test report.

**Comment by casparvl** [comment]
I've pushed it myself, hope you don't mind @stefan-wolfsheimer . This way we can start testing :)

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

**Comment by Micket** [comment]
Missing webengine. I'll build locally so i can easier investigate the logs. maybe it's python2 related

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3102.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/cf29c3957d43861dbbb5bc09e7229810 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.7, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/0dcb7e44c7bf5566cf0552b7271862b4 for a full test report.

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 12 out of 12 (1 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/e0431fe7eb91487df5cb28b78eb10c81 for a full test report.

**Comment by casparvl** [comment]
Well that's fun... same EasyConfig, it builds fine for me. I've checked, and `libQt5WegEngine.so` is _really_ there in the installation. The WebEngine related part of the configure output looks like this:

```
Qt WebEngine Build Tools:
  Use System Ninja ....................... yes
  Jumbo Build Merge Limit ................ 8
  Developer build ........................ no
  Sanitizer .............................. no
  QtWebEngine required system libraries:
    fontconfig ........................... yes
    dbus ................................. yes
    nss .................................. yes
    khr .................................. yes
    glibc ................................ yes
  QtWebEngine required system libraries for qpa-xcb:
    x11 .................................. yes
    libdrm ............................... yes
    xcomposite ........................... yes
    xcursor .............................. yes
    xi ................................... yes
    xproto (glproto) ..................... yes
    xtst ................................. yes
    xkbfile .............................. yes
  Optional system libraries used:
    re2 .................................. no
    icu .................................. no
    libwebp, libwebpmux and libwebpdemux . no
    opus ................................. no
    ffmpeg ............................... no
    libvpx ............................... no
    snappy ............................... yes
    glib ................................. yes
    zlib ................................. yes
    minizip .............................. no
    libevent ............................. yes
    libxml2 and libxslt .................. no
    lcms2 ................................ no
    png .................................. yes
    JPEG ................................. yes
    harfbuzz ............................. yes
    freetype ............................. yes
Qt WebEngineCore:
  Embedded build ......................... no
  Full debug information ................. no
  Pepper Plugins ......................... yes
  Printing and PDF ....................... yes
  Proprietary Codecs ..................... no
  Spellchecker ........................... yes
  Native Spellchecker .................... no
  WebRTC ................................. yes
  PipeWire over GIO ...................... no
  Geolocation ............................ yes
  WebChannel support ..................... yes
  Kerberos Authentication ................ no
  Extensions ............................. yes
  Support qpa-xcb ........................ yes
  Use ALSA ............................... no
  Use PulseAudio ......................... no
Qt WebEngineQml:
  Support Qt WebEngine Qml ............... yes
  UI Delegates ........................... yes
  Test Support ........................... no
Qt WebEngineWidgets:
  Support Qt WebEngine Widgets ........... yes
```
Any chance either of you can paste the config output from your failing build? It might point us to a system dependency that is picked up on my system, but not on yours...

**Comment by Micket** [comment]
> FYI, we do have a system `python2` command. I don't know if all of you do? Could be that gets picked up I guess...

yes it will

```
WARNING: Python version 2 (2.7.5 or later) is required to build QtWebEngine.

WARNING: Python version 2 (2.7.5 or later) is required to build QtPdf.
```


I looked to see if there was any hope of patching, but oh man, it's the entire chromium fork. Not fixed in qtwebengine 5.15.15 either:
https://invent.kde.org/qt/qt/qtwebengine/-/blob/5.15.15/configure.pri#L9

I'm pretty sure this will never be fixed for version 5. Grepping through qtwebengine/3rdparty/chromium there is like 100 scripts that are all python 2.

Lets start using Qt6 for things as much as possible instead. There are some changes, but not all that much, so some software might just work as is.

Add python2 as a builddep here.

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/1cc963e08c7eb0d68fe0ddc6b538c850 for a full test report.

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
vera-gpu1 - Linux Rocky Linux 8.7, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 525.85.12, Python 3.6.8
See https://gist.github.com/Micket/3150c44f80f01c47fd479b82f066dea0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node3103.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/8e8be4c5b97f62f6f8a23cddf48b550a for a full test report.

**Review by casparvl — APPROVED** [review]

**Comment by casparvl** [comment]
Going in, thanks @stefan-wolfsheimer!

**Pull request merged**